google-cloud-spanner/src/main/java/com/google/cloud/spanner/Value.java

```diff
@@ -1492,6 +1610,46 @@ int valueHash() {
     }
   }
 
+  private static class Float32Impl extends AbstractValue {
+    private final float value;
+
+    private Float32Impl(boolean isNull, float value) {
+      super(isNull, Type.float32());
+      this.value = value;
+    }
+
+    @Override
+    public float getFloat32() {
+      checkNotNull();
+      return value;
+    }
+
+    @Override
+    com.google.protobuf.Value valueToProto() {
+      return com.google.protobuf.Value.newBuilder().setNumberValue((double) value).build();
```

Comment:
nit: remove unnecessary cast

```suggestion
      return com.google.protobuf.Value.newBuilder().setNumberValue(value).build();
```